### PR TITLE
C++ constructors, resolves #298

### LINKED ISSUE #298: [Entry Request] C++: Constructors
## Description

Write a high-quality "Constructors" entry in C++. 

C++ topic: https://www.codecademy.com/resources/docs/cpp

## Type of Bug Request

- Typo and/or bug within one or more existing entries.
- Unclear explanation of term(s).
- Out-of-date information (new release, deprecated package, etc).
- Updating the documentation

## Type of Entry Request

- One or more new entries/terms requested.
- Recommend change(s) to documentation (such as existing text or new text)

## Checklist

- [x] The issue is appropriately titled.
- [x] Description is clear and includes supporting text, files, etc.
- [x] A similar issue hasn't already been created.
- [x] No more than 3 entries requested.

## For Maintainers

- [x] Labels are added
- [x] Assignee is added

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

Created a new entry for C++ constructors: docs/content/cpp/concepts/constructors/constructors.md

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/cpp/concepts/constructors/constructors.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+Title: 'Constructors'
+Description: 'A constructor is a specially defined method in a c++ class that is automatically called when an instance of that class is created.'
```
Comment (yangc95) [inline content/cpp/concepts/constructors/constructors.md:3]:
```suggestion
Description: 'A constructor is a specially defined method in a c=C++ class that is automatically called when an instance of that class is created.'
```

File: content/cpp/concepts/constructors/constructors.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+### Syntax and Examples
+
+A constructor method is defined within a class by using the class name followed by parenthesis:
+
+```cpp
+class MyClass {
+  public:
+    MyClass() {    // This defines a constructor that outputs a message
+      cout << "The constructor was executed!";
```
Comment (yangc95) [inline content/cpp/concepts/constructors/constructors.md:27]:
```suggestion
      std::cout << "The constructor was executed!";
```

Should we be using std::cout here?

File: content/cpp/concepts/constructors/constructors.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+int main() {
+  // Call the constructor using arguments
+  Circle myCircle1(5,5,10);
+  Circle myCircle2(0,0,5);
+
+  // Print out the attribute values set for each object
+  cout << "X=" << myCircle1.X << ", Y=" << myCircle1.Y << ", radius=" << myCircle1.radius << "\n";
+  cout << "X=" << myCircle2.X << ", Y=" << myCircle2.Y << ", radius=" << myCircle2.radius << "\n";
```
Comment (yangc95) [inline content/cpp/concepts/constructors/constructors.md:59]:
```suggestion
  std::cout << "X=" << myCircle1.X << ", Y=" << myCircle1.Y << ", radius=" << myCircle1.radius << "\n";
  std::cout << "X=" << myCircle2.X << ", Y=" << myCircle2.Y << ", radius=" << myCircle2.radius << "\n";
```

File: content/cpp/concepts/constructors/constructors.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  return 0;
+}
```
Comment (yangc95) [inline content/cpp/concepts/constructors/constructors.md:62]:
```suggestion
}

// Output: 
// X=5, Y=5, radius=10
// X=0, Y=0, radius=5
```

**Review by yangc95 — COMMENTED** [review]
@SSwiniarski I've done the first review of this entry. Very clear explanations, it was a good read. 📖

**Comment by SSwiniarski** [comment]
@yangc95, I applied all your suggestions. Ready for second review.

**Review by yangc95 — APPROVED** [review]

**Review by KTom101 — COMMENTED** [review]
@SSwiniarski Very nice entry! Only two small edits.

**Comment by SSwiniarski** [comment]
Suggestions applied. Should be good to go.

**Comment by KTom101** [comment]
Yep, everything looks good! Merging!

**Comment by KTom101** [comment]
It's live! https://www.codecademy.com/resources/docs/cpp/constructors

**Pull request merged**